src/main/java/teammates/common/util/Version.java

```diff
@@ -0,0 +1,152 @@
+            list = versionInString.split("-", 3);   // split into at most 3 parts
+        } else {
+            list = versionInString.split(".", 3);
+        }
+        if (list.length > 0) {
+            major = Integer.parseInt(list[0]);
+        }
+        if (list.length > 1) {
+            minor = Integer.parseInt(list[1]);
+        }
+        if (list.length > 2) {
+            patch = list[2];
+        }
+    }
+    
+    /**
+     * Compares by string representation.
+     */
+    public boolean equals(Object anotherVersion) {
+        return toString().equals(anotherVersion.toString());
+    }
+    
+    /**
+     * Gets hash code for this version.
+     */
+    public int hashCode() {
+        return toString().hashCode();
+    }
+    
+    /**
+     * Converts Version to String in format XX.XX.XXXX
+     */
+    public String toString() {
+        return originalRepresentation.replaceAll("-", ".");
+    }
+    
+    /**
+     * Converts to String in format XX-XX-XXXX
+     */
+    public String toStringForQuery() {
+        return originalRepresentation.replaceAll(".", "-");
+    }
+    
+    private int compareVersionNumber(Integer num1, Integer num2) {
+        if (num1 == null && num2 == null) {
+            return 0;
+        }
+        if (num1 == null) {
+            return 1;
+        }
+        if (num2 == null) {
+            return -1;
+        }
+        return -num1.compareTo(num2);
+    }
+    
+    private int compareVersionString(String s1, String s2) {
+        if (s1 == null && s2 == null) {
+            return 0;
+        }
+        if (s1 == null) {
+            return 1;
+        }
+        if (s2 == null) {
+            return -1;
+        }
+        return -s1.compareTo(s2);
+    }
+    
+    /**
+     * Compares versions by major, minor then by patch.
+     * The version with greater major, minor or patch will be smaller.
+     */
+    @Override
+    public int compareTo(Version anotherVersion) {
+        int majorComparisonResult = compareVersionNumber(this.getMajorVersion(), anotherVersion.getMajorVersion());
+        if (majorComparisonResult != 0) {
+            return majorComparisonResult;
+        }
+        int minorComparisonResult = compareVersionNumber(this.getMinorVersion(), anotherVersion.getMinorVersion());
+        if (minorComparisonResult != 0) {
+            return minorComparisonResult;
+        }
+        int patchComparisonResult = compareVersionString(this.getPatchVersion(), anotherVersion.getPatchVersion());
+        return patchComparisonResult;
+    }
+    
+    /**
+     * Gets the major version number of the version.
+     */
+    public Integer getMajorVersion() {
+        return major;
+    }
+    
+    /**
+     * Gets the minor version number of the version.
+     */
+    public Integer getMinorVersion() {
+        return minor;
+    }
+    
+    /**
+     * Gets the patch part of the version.
+     */
+    public String getPatchVersion() {
+        return patch;
+    }
```

Comment:
Are the getters used outside the class? If they're not, I don't really see the need for getters because you can very simply call `version.major/minor/patch`.